stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -33,18 +46,276 @@ public void disable(JsonRpcPeer peer, JSONObject params) {
   }
 
   @ChromeDevtoolsMethod
-  public JsonRpcResult getSupportedCSSProperties(JsonRpcPeer peer, JSONObject params) {
-    GetSupportedCSSPropertiesResponse response = new GetSupportedCSSPropertiesResponse();
-    response.cssProperties = Collections.emptyList();
-    return response;
+  public JsonRpcResult getComputedStyleForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetComputedStyleForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetComputedStyleForNodeRequest.class);
+
+    final GetComputedStyleForNodeResult result = new GetComputedStyleForNodeResult();
+    result.computedStyle = new ArrayList<>();
+
+    mDocument.postAndWait(new Runnable() {
+      @Override
+      public void run() {
+        mDocument.getElementStyles(
+            mDocument.getElementForNodeId(request.nodeId),
+            new StyleAccumulator() {
+              @Override
+              public void store(String name, String value, boolean isDefault) {
+                if (!isDefault) {
+                  CSSComputedStyleProperty property = new CSSComputedStyleProperty();
+                  property.name = name;
+                  property.value = value;
+
+                  result.computedStyle.add(property);
+                }
+              }
+            });
+      }
+    });
+
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params) {
+    final GetMatchedStylesForNodeRequest request = mObjectMapper.convertValue(
+        params,
+        GetMatchedStylesForNodeRequest.class);
+
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+
+    result.matchedCSSRules = new ArrayList<>();
```

Comment:
if we know we're only going to have 1 item in this list, we can specialize this. First idea would be to make sure we do something like `new ArrayList<>(1);` (specify capacity). But, we can do better because that still requires 2 allocations (one for the ArrayList, one for its internal array). 

In this case, because we're not modifying the list, we can add a method to `ListUtil` for creating a special 1-item `List` implementation, similar to what we're doing with the "immutable" lists. In fact, really what we need is a `ListUtil.newImmutableList(T)` method,

```
public static <T> newImmutableList(T item) {
  return new OneItemImmutableList<>(item);
}
```